src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -1425,6 +1425,118 @@ void CVulkanLogicalDevice::createGraphicsPipelines_impl(
+    core::vector<VkRayTracingShaderGroupCreateInfoKHR> vk_shaderGroup(maxShaderStages, { VK_STRUCTURE_TYPE_RAY_TRACING_SHADER_GROUP_CREATE_INFO_KHR, nullptr});
+    core::vector<VkSpecializationInfo> vk_specializationInfos(maxShaderStages, { 0, nullptr, 0, nullptr });
+    core::vector<VkSpecializationMapEntry> vk_specializationMapEntry(validation.count);
+    core::vector<uint8_t> specializationData(validation.dataSize);
+
+    auto outCreateInfo = vk_createInfos.data();
+    auto outRequiredSubgroupSize = vk_requiredSubgroupSize.data();
+    auto outShaderStage = vk_shaderStage.data();
+    auto outShaderGroup = vk_shaderGroup.data();
+    auto outSpecInfo = vk_specializationInfos.data();
+    auto outSpecMapEntry = vk_specializationMapEntry.data();
+    auto outSpecData = specializationData.data();
+    auto getVkShaderIndex = [](uint32_t index) { return index == SShaderGroupParams::ShaderUnused ? VK_SHADER_UNUSED_KHR : index;  };
+    auto getGeneralVkRayTracingShaderGroupCreateInfo = [getVkShaderIndex](SGeneralShaderGroup group) -> VkRayTracingShaderGroupCreateInfoKHR
+    {
+        return {
+            .sType = VK_STRUCTURE_TYPE_RAY_TRACING_SHADER_GROUP_CREATE_INFO_KHR,
+            .pNext = nullptr,
+            .type = VK_RAY_TRACING_SHADER_GROUP_TYPE_GENERAL_KHR,
+            .generalShader = getVkShaderIndex(group.shaderIndex),
+            .closestHitShader = VK_SHADER_UNUSED_KHR,
+            .anyHitShader = VK_SHADER_UNUSED_KHR,
+            .intersectionShader = VK_SHADER_UNUSED_KHR,
+        };
+    };
+    auto getHitVkRayTracingShaderGroupCreateInfo = [getVkShaderIndex](SHitShaderGroup group) -> VkRayTracingShaderGroupCreateInfoKHR
+    {
+        return  {
+            .sType = VK_STRUCTURE_TYPE_RAY_TRACING_SHADER_GROUP_CREATE_INFO_KHR,
+            .pNext = nullptr,
+            .type = group.intersectionShaderIndex == SShaderGroupParams::ShaderUnused ? 
+              VK_RAY_TRACING_SHADER_GROUP_TYPE_TRIANGLES_HIT_GROUP_KHR : VK_RAY_TRACING_SHADER_GROUP_TYPE_PROCEDURAL_HIT_GROUP_KHR,
+            .generalShader = VK_SHADER_UNUSED_KHR,
+            .closestHitShader = getVkShaderIndex(group.closestHitShaderIndex),
+            .anyHitShader = getVkShaderIndex(group.anyHitShaderIndex),
+            .intersectionShader = getVkShaderIndex(group.intersectionShaderIndex),
+        };
+    };
+    for (const auto& info : createInfos)
+    {
+        initPipelineCreateInfo(outCreateInfo,info);
+        outCreateInfo->pStages = outShaderStage;
+        for (const auto& specInfo : info.shaders)
+        {
+            if (specInfo.shader)
+            {
+                *(outShaderStage++) = getVkShaderStageCreateInfoFrom(specInfo,outRequiredSubgroupSize,outSpecInfo,outSpecMapEntry,outSpecData);
+            }
+        }
+        outCreateInfo->stageCount = std::distance<decltype(outCreateInfo->pStages)>(outCreateInfo->pStages,outShaderStage);
```

Comment:
I think its probably not valid to create a RT pipeline with 0 shaders, so assert that `outCreateInfos->stageCount>0` (and also make sure to validate that before entering the function)